pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go

```diff
@@ -37,48 +38,64 @@ import (
 	"github.com/oam-dev/kubevela/pkg/oam/util"
 )
 
-// Reconciler reconciles a WorkloadDefinition object
+// Reconciler reconciles a ComponentDefinition object
 type Reconciler struct {
 	client.Client
 	dm     discoverymapper.DiscoveryMapper
 	Scheme *runtime.Scheme
 	record event.Recorder
 }
 
-// Reconcile is the main logic for WorkloadDefinition controller
+// Reconcile is the main logic for ComponentDefinition controller
 func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	definitionName := req.NamespacedName.Name
-	if definitionName == "containerizedworkloads.core.oam.dev" {
+	klog.InfoS("Reconciling ComponentDefinition", "Name", definitionName, "Namespace", req.Namespace)
+	ctx := context.Background()
+
+	var componentDefinition v1alpha2.ComponentDefinition
+	if err := r.Get(ctx, req.NamespacedName, &componentDefinition); err != nil {
+		if kerrors.IsNotFound(err) {
+			err = nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// TODO: check finalizer
+	if componentDefinition.DeletionTimestamp != nil {
 		return ctrl.Result{}, nil
 	}
 
-	klog.InfoS("Reconciling WorkloadDefinition", "Name", definitionName, "Namespace", req.Namespace)
-	ctx := context.Background()
-	var def utils.CapabilityWorkloadDefinition
+	fmt.Println("this step is running")
+	var def utils.CapabilityComponentDefinition
 	def.Name = req.NamespacedName.Name
-
 	err := def.StoreOpenAPISchema(ctx, r, req.Namespace, req.Name)
 	if err != nil {
 		klog.ErrorS(err, "cannot store capability in ConfigMap")
-		r.record.Event(&(def.WorkloadDefinition), event.Warning("cannot store capability in ConfigMap", err))
-		// TODO(zzxwill) The error message should also be patched into Status
-		return ctrl.Result{}, util.PatchCondition(ctx, r, &(def.WorkloadDefinition),
+		r.record.Event(&(def.ComponentDefinition), event.Warning("cannot store capability in ConfigMap", err))
+		// TODO(yangsoon) The error message should also be patched into Status
+		return ctrl.Result{}, util.PatchCondition(ctx, r, &(def.ComponentDefinition),
 			cpv1alpha1.ReconcileError(fmt.Errorf(util.ErrStoreCapabilityInConfigMap, def.Name, err)))
 	}
+
+	if err := r.Update(ctx, &def.ComponentDefinition); err != nil {
```

Comment:
what are we updating, do you mean update status, then we should use sub resource update.